flink-filesystems/flink-hadoop-fs/src/main/java/org/apache/flink/runtime/fs/hdfs/HadoopRecoverableWriter.java

```diff
@@ -60,6 +60,12 @@ public HadoopRecoverableWriter(org.apache.hadoop.fs.FileSystem fs) {
 	@Override
 	public RecoverableFsDataOutputStream open(Path filePath) throws IOException {
 		final org.apache.hadoop.fs.Path targetFile = HadoopFileSystem.toHadoopPath(filePath);
+
+		// the finalized part must not exist already
+		if (fs.exists(targetFile)) {
```

Comment:
This check should also exist before. It essentially says that we should never try to create a file with the same part name as an already existing file that is finalised/committed. In other case, this would overwrite committed data.